### PR TITLE
Removed BinaryCompatbility from DirectoryServices

### PR DESCRIPTION
**Pull request by ViktorHofer**
fixes https://github.com/dotnet/corefx/issues/15810

### CODE REVIEW

File: src/System.DirectoryServices/src/System.DirectoryServices.csproj [hunk 1]
```diff
@@ -17,12 +17,15 @@
     <Compile Include="System\DirectoryServices\ActiveDirectorySecurity.cs" />
     <Compile Include="System\DirectoryServices\AdsVLV.cs" />
     <Compile Include="System\DirectoryServices\AuthenticationTypes.cs" />
-    <Compile Include="System\DirectoryServices\BinaryCompatibility.cs" />
     <Compile Include="System\DirectoryServices\DerefAlias.cs" />
     <Compile Include="System\DirectoryServices\DirectoryEntries.cs" />
-    <Compile Include="System\DirectoryServices\DirectoryEntry.cs" />
+    <Compile Include="System\DirectoryServices\DirectoryEntry.cs">
+      <SubType>Component</SubType>
```
Comment (jkotas) [inline src/System.DirectoryServices/src/System.DirectoryServices.csproj]: Why was this added?

Comment (ViktorHofer) [inline src/System.DirectoryServices/src/System.DirectoryServices.csproj]: This gets automatically added when you open the solution with Visual Studio and the class inherits from System.ComponentModel.Component. If you want I can undo this but it will occur again next time someone opens and builds it with VS.

**Comment by ViktorHofer** [comment]
@dotnet-bot test Innerloop Ubuntu14.04 arm Release Cross Build
@dotnet-bot test Innerloop Ubuntu16.04 arm Debug Cross Build

File: src/System.DirectoryServices/src/System.DirectoryServices.csproj (hunk 1 above)
Comment (jkotas) [inline src/System.DirectoryServices/src/System.DirectoryServices.csproj]: Keep it. I was just wondering whether it was intentional.

**Comment by ViktorHofer** [comment]
@dotnet-bot test Innerloop Ubuntu16.04 arm Debug Cross Build

**Comment by jkotas** [comment]
arm build is broken right now
